src/parsers/manifest/smooth/representation_index.ts

```diff
@@ -258,53 +258,33 @@ export default class SmoothRepresentationIndex implements IRepresentationIndex {
    */
   getSegments(_up : number, _to : number) : ISegment[] {
     this._refreshTimeline();
-    const index = this._index;
-    const { up, to } = normalizeRange(index, _up, _to);
-    const { timeline, timescale, media } = index;
+    const { up, to } = normalizeRange(this._index, _up, _to);
+    const { timeline, timescale, media } = this._index;
+    const isAggressive = this._isAggressiveMode;
 
     let currentNumber : number|undefined;
     const segments : ISegment[] = [];
 
     const timelineLength = timeline.length;
 
-    // TODO(pierre): use @maxSegmentDuration if possible
-    let maxEncounteredDuration = (timeline.length && timeline[0].duration) || 0;
-
-    const maxPosition = this._isAggressiveMode || this._scaledLiveGap == null ?
-      undefined : ((performance.now() / 1000) * timescale) - this._scaledLiveGap;
+    const maxPosition = this._scaledLiveGap == null ?
+      undefined :
+      ((performance.now() / 1000) * timescale) - this._scaledLiveGap;
 
     for (let i = 0; i < timelineLength; i++) {
       const segmentRange = timeline[i];
       const { duration, start } = segmentRange;
-
-      maxEncounteredDuration = Math.max(maxEncounteredDuration, duration || 0);
-
-      // live-added segments have @d attribute equals to -1
-      if (duration != null && duration < 0) {
-        const approximateEnd = start + maxEncounteredDuration;
-        if (approximateEnd < to &&
-            (maxPosition == null || approximateEnd <= maxPosition))
-        {
-          const time = start;
-          const segment = { id: "" + time,
-                            time,
-                            isInit: false,
-                            timescale,
-                            number: currentNumber != null ? currentNumber :
-                                                            undefined,
-                            mediaURL: replaceSegmentSmoothTokens(media, time) };
-          segments.push(segment);
-        }
-        return segments;
-      }
-
```

Comment:
Removed that logic has it was tedious and as we removed that in the low-latency branch anyway.
To check if this has an effect.